nomos-services/data-availability/network/src/storage/adapters/mock.rs

```diff
@@ -0,0 +1,19 @@
+use std::{collections::HashMap, sync::Mutex};
+
+use crate::{membership::Assignations, storage::MembershipStorage};
+
+#[derive(Default)]
+pub struct MockStorage {
+    storage: Mutex<HashMap<u64, Assignations>>,
+}
+
+impl MembershipStorage for MockStorage {
+    fn store(&self, block_number: u64, assignations: Assignations) {
+        let mut storage = self.storage.lock().unwrap();
+        storage.insert(block_number, assignations);
+    }
+
+    fn get(&self, block_number: u64) -> Option<Assignations> {
+        self.storage.lock().unwrap().get(&block_number).cloned()
+    }
+}
```

Comment:
If this have the store it means the network service is responsible of storing it. I know we had this convo already, it can be done here, but somehow it feels a bit sketchy.

Reply:
Hmm, this makes sense. Let's see if this glue code is ok, and I can try to refactor storage later. I think it will be easier to review :)